lib/network.go

```diff
@@ -857,11 +857,12 @@ func (msg *MsgDeSoGetBlocks) String() string {
 	return fmt.Sprintf("%v", msg.HashList)
 }
 
-// Within a post, the body typically has a particular
+// DeSoBodySchema Within a post, the body typically has a particular
 // schema defined below.
 type DeSoBodySchema struct {
-	Body      string
-	ImageURLs []string
+	Body      string `json:",omitempty"`
+	ImageURLs []string `json:",omitempty"`
+	VideoURLs []string `json:",omitempty"`
```

Comment:
I think this change is acceptable, but would like other core devs (@redpartyhat, @diamondhands0 , @maebeam ) to weigh in. Omitting VideoURLs when they are empty allow us to avoid a fork as existing posts can still be marshaled and unmarshaled in the same fashion since it won't expect the VideoURLs.

I think adding omitempty to the other attributes is okay as well and will save us a little space in the long run as we won't have keys in body if there is no value.

Reply:
So, just to clarify my understanding... I don't think adding VideoURLs to the DeSoBodySchema would actually cause a fork with existing forks.  If Golang tries to unmarshal a JSON string that is missing a key, it just initializes the struct with the default value.  Therefore, old posts wouldn't actually break.  See here for an example of unmarshalling a JSON string that is missing a key: https://play.golang.org/p/e4QnZgGpsh5. Let me know if I am missing something and it would in fact cause a fork.  

That being said, I see no reason not to omit empty values when storing these JSON blobs on the chain! This change LGTM.

Reply:
Okay that makes sense - my understanding of how it would marshal when missing a key was incorrect. If you're good with this change, then let's do it! I synced a whole node from scratch with this update, so it won't mess up the existing chain.